src/idpi/grib_decoder.py

```diff
@@ -331,17 +321,22 @@ def load(
             Mapping of fields by param name
 
         """
-        params = set()
-        for desc in descriptors:
-            params |= set(desc.input_fields)
+        reqs = {req for desc in descriptors for req in desc.input_fields}
+
+        return self.load_fields(reqs, extract_pv=extract_pv)
 
-        return self.load_fields(params, extract_pv=extract_pv)
+    def load_fieldnames(
+        self, params: list[str], extract_pv: str | None = None
```

Comment:
Formatting of args is different for this function. also can we have a docstring? There are now a few functions with very similar names/arguments so its not so clear whats what on first glance.